compiler/src/test/java/com/squareup/anvil/compiler/codegen/BindingModuleGeneratorTest.kt

```diff
@@ -0,0 +1,205 @@
+import com.squareup.anvil.annotations.MergeComponent
+import com.squareup.anvil.annotations.MergeSubcomponent
+import com.squareup.anvil.annotations.compat.MergeModules
+import com.squareup.anvil.compiler.AnyDaggerComponent
+import com.squareup.anvil.compiler.MODULE_PACKAGE_PREFIX
+import com.squareup.anvil.compiler.anyDaggerComponent
+import com.squareup.anvil.compiler.compile
+import com.squareup.anvil.compiler.componentInterface
+import com.squareup.anvil.compiler.componentInterfaceAnvilModule
+import com.squareup.anvil.compiler.contributingInterface
+import com.squareup.anvil.compiler.daggerModule
+import com.squareup.anvil.compiler.daggerModule1
+import com.squareup.anvil.compiler.parentInterface
+import com.squareup.anvil.compiler.subcomponentInterface
+import com.squareup.anvil.compiler.subcomponentInterfaceAnvilModule
+import com.tschuchort.compiletesting.KotlinCompilation.ExitCode.COMPILATION_ERROR
+import dagger.Binds
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+import java.lang.reflect.Modifier.isAbstract
+import kotlin.reflect.KClass
+
+@RunWith(Parameterized::class)
+class BindingModuleGeneratorTest(
+  private val annotationClass: KClass<*>
+) {
+
+  private val annotation = "@${annotationClass.simpleName}"
+  private val import = "import ${annotationClass.java.canonicalName}"
+
+  companion object {
+    @Parameters(name = "{0}")
+    @JvmStatic fun annotationClasses(): Collection<Any> {
+      return listOf(MergeComponent::class, MergeSubcomponent::class, MergeModules::class)
+    }
+  }
+
+  @Test fun `a Dagger module is generated for a merged class and added to the component`() {
+    compile(
+        """
+        package com.squareup.test
+        
+        $import
+        
+        $annotation(Any::class)
+        interface ComponentInterface
+    """
+    ) {
+      val modules = if (annotationClass == MergeModules::class) {
+        componentInterface.daggerModule.includes.toList()
+      } else {
+        componentInterface.anyDaggerComponent.modules
+      }
+      assertThat(modules).containsExactly(componentInterfaceAnvilModule.kotlin)
+    }
+  }
+
+  @Test fun `the Anvil module is merged with other modules`() {
+    compile(
+        """
+        package com.squareup.test
+        
+        import com.squareup.anvil.annotations.ContributesTo
+        $import
+        
+        @ContributesTo(Any::class)
+        @dagger.Module
+        abstract class DaggerModule1
+        
+        $annotation(Any::class)
+        interface ComponentInterface
+    """
+    ) {
+      val modules = if (annotationClass == MergeModules::class) {
+        componentInterface.daggerModule.includes.toList()
+      } else {
+        componentInterface.anyDaggerComponent.modules
+      }
+      assertThat(modules)
+          .containsExactly(componentInterfaceAnvilModule.kotlin, daggerModule1.kotlin)
```

Comment:
Does expected order matter here? If yes, might need to add `.inOrder()`.

Reply:
No, the order doesn't matter for Dagger.